analyzers/src/SonarAnalyzer.CSharp/Rules/CollectionQuerySimplification.cs

```diff
@@ -30,10 +30,17 @@ public sealed class CollectionQuerySimplification : SonarDiagnosticAnalyzer
         internal const string MessageDropFromMiddle = "Drop this useless call to '{0}' or replace it by 'AsEnumerable' if " +
             "you are using LINQ to Entities.";
 
-        private static readonly DiagnosticDescriptor rule =
+        private static readonly DiagnosticDescriptor Rule =
             DescriptorFactory.Create(DiagnosticId, MessageFormat);
 
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(rule);
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(Rule);
+
+        private static readonly ImmutableArray<KnownType> DatabaseLinqBaseTypes =
```

Comment:
I think this collection comes in handy in other places as well. We should consider adding it to KnownTypes directly.